### PR TITLE
npcs from dict to list + iids

### PR DESCRIPTION
**Pull request by JaskRendix**
PR proposes to change **self.npcs** from dict to list; it adds also a method to get a NPC based on his instance_id

### CODE REVIEW

File: tuxemon/event/__init__.py [hunk 1]
```diff
@@ -65,10 +71,29 @@ def get_npc(session: Session, slug: str) -> Optional[NPC]:
...
+
+    Parameters:
+        session: The session object.
+        iid: The iid of the NPC that exists on the current map.
+
+    Returns:
+        The NPC object or None if the NPC is not found.
+
+    """
+    from tuxemon.states.world.worldstate import WorldState
```
Comment (Grimmys) [inline tuxemon/event/__init__.py:89]: Why is the import done there instead of at the top of the file?

File: tuxemon/event/__init__.py [hunk 2]
```diff
@@ -65,10 +71,29 @@ def get_npc(session: Session, slug: str) -> Optional[NPC]:
...
+
+    # Loop through the NPC list and see if the slug matches any in the list
```
Comment (Grimmys) [inline tuxemon/event/__init__.py]: Isn't this comment only describing what's happening inside of  `world.get_entity_by_iid`?
If so, I'm not sure if it's useful there.

**Review by Grimmys — COMMENTED** [review]
Interesting changes, looks like the change from `dict` to `list` is simplifying a lot the implementation!
Thanks.

File: tuxemon/event/__init__.py (hunk 1 above)
Comment (JaskRendix) [reply tuxemon/event/__init__.py:89]: No idea, I found like this, I try to move it

Comment (JaskRendix) [reply tuxemon/event/__init__.py:89]: (most likely due to a circular import)

File: tuxemon/event/__init__.py (hunk 2 above)
Comment (JaskRendix) [reply tuxemon/event/__init__.py]: good catch

**Review by Grimmys — APPROVED** [review]

**Review by bitcraft — APPROVED** [review]

**Pull request merged**